src/topology/index.js

```diff
@@ -62,6 +62,9 @@ class Topology {
     return Boolean(other && other[topologySymbol])
   }
 
+  /**
+   * @param {any} registrar
```

Comment:
We would need to create a circular dependency to have registrar types in the interface. We can change this interaction, but I feel we should not spend a lof of time reinventing this interaction at the moment.